src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/SubmissionInterceptor.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.companieshouse.confirmationstatementapi.interceptor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.confirmationstatementapi.exception.SubmissionNotFoundException;
+import uk.gov.companieshouse.confirmationstatementapi.service.ConfirmationStatementService;
+import uk.gov.companieshouse.confirmationstatementapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.*;
+
+@Component
+public class SubmissionInterceptor implements HandlerInterceptor {
+
+    private final ConfirmationStatementService confirmationStatementService;
+
+    @Autowired
+    public SubmissionInterceptor(ConfirmationStatementService confirmationStatementService) {
+        this.confirmationStatementService = confirmationStatementService;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        final Map<String, String> pathVariables = (Map<String, String>) request
+                .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+        final var submissionId = pathVariables.get(CONFIRMATION_STATEMENT_ID_KEY);
+
+        var logMap = new HashMap<String, Object>();
+        String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
+        try {
+            boolean isPresent = confirmationStatementService.getConfirmationStatement(submissionId).isPresent();
+            ApiLogger.debugContext(reqId, "Submission found", logMap);
```

Comment:
if it is present, no harm in adding the submission to the request as an attribute. (The transaction interceptor does this)
I don't think anything needs it right now, but as we've gone through the effort of retrieving it, may as well. 
Then anything requiring the submission to be present won't need to do another db call